robot/impl/robot_reconfigure_test.go

```diff
@@ -3588,119 +3584,100 @@ func TestReconfigureRename(t *testing.T) {
 	test.That(t, res2.(*mockFake).closeCount, test.ShouldEqual, 0)
 }
 
+// tests that the resource configuration timeout is passed into each resource constructor.
 func TestResourceConstructTimeout(t *testing.T) {
-	cfg := &config.Config{}
-	ctx := context.Background()
-	logger, logs := logging.NewObservedTestLogger(t)
-	fakeModel := resource.DefaultModelFamily.WithModel("fake")
-
-	timeOutErrorCount := func() int {
-		return logs.Filter(func(o observer.LoggedEntry) bool {
-			return strings.Contains(
-				o.Entry.Message,
-				"resource build error: resource rdk:component:base/fakewheel timed out after 1ns during reconfigure")
-		}).Len()
-	}
+	logger := logging.NewTestLogger(t)
 
-	r, err := New(ctx, cfg, logger)
-	defer func() {
-		test.That(t, r.Close(context.Background()), test.ShouldBeNil)
-	}()
-	test.That(t, err, test.ShouldBeNil)
+	mockAPI := resource.APINamespaceRDK.WithComponentType("mock")
+	modelName1 := utils.RandomAlphaString(5)
+	model1 := resource.DefaultModelFamily.WithModel(modelName1)
+
+	var timeout time.Duration
 
-	// test no error logging with default config
-	testutils.WaitForAssertion(t, func(tb testing.TB) {
-		test.That(tb, timeOutErrorCount(), test.ShouldEqual, 0)
+	resource.RegisterComponent(mockAPI, model1, resource.Registration[resource.Resource, resource.NoNativeConfig]{
+		Constructor: func(
+			ctx context.Context,
+			deps resource.Dependencies,
+			conf resource.Config,
+			logger logging.Logger,
+		) (resource.Resource, error) {
+			deadline, ok := ctx.Deadline()
+			test.That(t, ok, test.ShouldBeTrue)
+			test.That(t, deadline, test.ShouldHappenBetween, time.Now().Add(timeout/2), time.Now().Add(timeout))
```

Comment:
> test should really only care if the deadline is passed correctly into the constructor, so updated the test to be more centered around that

That's clever! Definitely the right way to be testing the semantics here.

A note about the assertion as written (not a mandate to change anything):
I don't think this test requires the rigor to avoid false positives given the time scales at play. But my intuition would be to merely test that:
```
time.Now() + construct timeout <= deadline
```
That assertion only relies on clocks moving forward. But the one written makes assumptions about process/goroutine scheduling.

Reply:
updated, thanks for the idea! I was trying to think of a better way to test it but didn't spend that much time